ui/components/app/transaction-decoding/components/decoding/address/address.component.js

```diff
@@ -19,16 +22,29 @@ const Address = ({
   const [showNicknamePopovers, setShowNicknamePopovers] = useState(false);
 
   const addressBook = useSelector(getAddressBook);
-  const addressBookEntryObject = addressBook.find(
-    (entry) => entry.address === checksummedRecipientAddress,
+  const getNickName = (address) => {
+    const addressBookEntryObject = addressBook.find((entry) => {
+      return address.toLowerCase() === entry.address.toLowerCase();
+    });
+    return (addressBookEntryObject && addressBookEntryObject?.name) || '';
+  };
+
+  const recipientNickname = getNickName(checksummedRecipientAddress);
+  const recipientMetadataName = getMetadataContractName(
+    checksummedRecipientAddress,
   );
-  const recipientNickname = addressBookEntryObject?.name;
 
   const recipientToRender = addressOnly
-    ? recipientNickname ||
```

Comment:
Before this PR, `recipientName` was not one of the options if `addressOnly` was true. Now this PR adds it as one of the options in that case. Do we know why it wasn't one of the options previously and why this is an okay change to make?

Reply:
@danjm Yes, like as I mentioned in the above comment https://github.com/MetaMask/metamask-extension/pull/16018#discussion_r1003299116 I followed the AC in this ticket: https://github.com/MetaMask/metamask-extension/issues/15780. 

I followed that for displaying this addresses and usages from address component. Let me know if this is the right approach for this ticket.

Thank you in advance!

Reply:
I think the solution is that we shouldn't use the `addressOnly` flag on the address component in `ui/components/app/signature-request/signature-request-message/signature-request-message.js`, and then we don't need to add `recipientName` to this list

Reply:
@danjm I will share here screenshots with different situations. 

1) Here I **used** `addressOnly` flag on the address component in `ui/components/app/signature-request/signature-request-message/signature-request-message.js` and here is **added** `recipientName` as one of the options in address component:

![Screenshot 2022-11-03 at 15 30 41](https://user-images.githubusercontent.com/92527393/199753759-578d4829-7af2-4f5e-9104-59014d6cbd15.png)

2) Here I **used** `addressOnly` flag on the address component in `ui/components/app/signature-request/signature-request-message/signature-request-message.js` and here is **not added** `recipientName` as one of the options in address component:

![Screenshot 2022-11-03 at 15 33 20](https://user-images.githubusercontent.com/92527393/199753983-d122ad03-e5e7-4831-bfdc-7a467d6946fa.png)

3) Here I **don't used** `addressOnly` flag on the address component in `ui/components/app/signature-request/signature-request-message/signature-request-message.js`:

![Screenshot 2022-11-03 at 15 45 32](https://user-images.githubusercontent.com/92527393/199754215-0bd7b88b-d1ca-43de-86fb-8230b27c2146.png)

So as you can see, **third case** is not valid because it will be displayed `New contract` instead of `shorten address` (it is not like on [Figma design](https://www.figma.com/file/U2Ojf3cJP4PvIYRiRfUya0/Improve-Signatures-(v1)?node-id=1971%3A1763) for resolving this ticket).

So if we want to display `shorten address` like on Figma design, we need to add `addressOnly` flag on the address component in `ui/components/app/signature-request/signature-request-message/signature-request-message.js`.

**Second case** is displaying `shorten address` instead of `account name` (in this case `Account 1`). `Account name` is displayed in the Figma design. 

You can see in this ticket https://github.com/MetaMask/metamask-extension/issues/15780 that `first case` in AC is this: `If the address exists in the user account list, then the account name is displayed (this should be the first thing checked, if it does exist in user account list we don't need to check the rest)`.

**First case** are displaying `account name` because `recipientName` is added as one of the options in address component. 

So I think we must used `addressOnly` flag on the address component in `ui/components/app/signature-request/signature-request-message/signature-request-message.js`, but I can remove `recipientName` as one of the options in address component, but than we will see only `shorten addresses` instead of `account name` if exists. In below screenshot:

![Screenshot 2022-11-03 at 16 08 44](https://user-images.githubusercontent.com/92527393/199758846-f687766d-7b21-4890-89f0-b3acdae11332.png)

`0xcF2...C59E` is my account name (Account 1).

Reply:
For the third case, where's `New contract` coming from? Is the nickname that you created within Contacts for this address? If yes, we would indeed like to display it here. 

From AC:
> If the address exists in the user address book (contacts), then their nickname is displayed (this should be already existing logic for this component - this should be second thing checked);

Reply:
Are we using the same component/logic on this other PR https://github.com/MetaMask/metamask-extension/pull/15888 ?

The way I see it, is that we would have a single component carries this logic of displaying account name > account nickname > contract metadata > address (the one described in AC) and it's used everywhere. So if we ever decide to change the logic, we only update code in one place and it get updated everywhere.

Reply:
Initially my thought was this ticket https://github.com/MetaMask/metamask-extension/issues/15780 would create the new logic for displaying address for this existing address component and this one https://github.com/MetaMask/metamask-extension/issues/15716 (the one we're currently discussing) would make the change to start using that same component within signTypedData_v4 messages. 

Does this makes sense? I was not able to fully follow the technical discussion above around flags and components.

Reply:
> For the third case, where's New contract coming from? Is the nickname that you created within Contacts for this address? If yes, we would indeed like to display it here.

@bschorchit `New contract` is coming from translations `t('newContract')`, it is not the nickname that I created within Contacts for this address.

Reply:
Forgive me for adding noise but for option 3 displaying a new contract. Is there any benefit in displaying the condensed contract address and using some tag to notify the user that it is a new contract?

Here is a rough mock I made with terrible copy

![new contract mock](https://user-images.githubusercontent.com/8112138/200361666-7648e5a2-b3c2-4461-a944-837b97ca47e4.png)

Reply:
Very interesting @georgewrmarshall! @SayaGT was exploring some designs in the past for showing address insights to users. Let's revisit this again soon.